curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641da803d9892447d059804e.md

```diff
@@ -0,0 +1,170 @@
+---
+id: 641da803d9892447d059804e
+title: Step 18
+challengeType: 0
+dashedName: step-18
+---
+
+# --description--
+
+Now create an image tag and give it the `class` `user-img`. Use template interpolation to set the `src` attribute to `image` you destructured earlier. Set the `alt` attribute to `author` followed by the text `avatar`. Make sure there is a space between the `author` variable and the word `avatar`, for example, `Quincy Larson avatar`.
+
+# --hints--
+
+You should create an `img` element.
+
+```js
+assert.exists(document.querySelector('img'));
+```
+
+Your `img` element should have the class `user-img`.
+
+```js
+assert.match(code, /<img\s*class=('|")user-img\1\s*/);
+```
+
+You should set the `src` attribute of your `img` element to `${image}`.
+
+```js
+assert.match(code, /<img\s*class=('|")user-img\1\s*src=\1\$\{image\}\1/)
+```
+
+You should set the `alt` attribute of your `img` element to `${author} avatar`.
+
+```js
+assert.match(code, /<img\s*class=('|")user-img\1\s*src=\1\$\{image\}\1\s+alt=\1\$\{author\}\s+avatar\1\/>/)
+```
```

Comment:
````suggestion
You should create an `img` element.

```js
assert.exists(document.querySelector('img'));
```

Your `img` element should have the class `user-img`.

```js
assert.include(document.querySelector('img')?.className, "user-img");
```

You should set the `src` attribute of your `img` element to `${image}`.

```js
assert.equal(document.querySelector('img')?.getAttribute('src'), authorDataArr[0].image);
```

You should set the `alt` attribute of your `img` element to `${author} avatar`.

```js
assert.equal(document.querySelector('img')?.getAttribute('alt'), `${authorDataArr[0].author} avatar`);
```
````

DOM queries save us from worrying about the order of attributes.